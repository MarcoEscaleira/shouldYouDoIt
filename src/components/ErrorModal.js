import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {actions} from "../store/decisions";

const ErrorModal = ({ show, text, closeTypeError}) => (
  <div className={`modal ${!show ? 'modal--invisible' : ''}`}>
    <div className="modal__overlay" onClick={closeTypeError}>
      <div className="modal__content">
        <h3 className="modal__title">Error</h3>
        <p className="modal__body">{text}</p>
        <button 
          className="modal__button"
          onClick={closeTypeError}
        > 
          Okay
        </button>
      </div>
    </div>
  </div>
);

ErrorModal.propTypes = {
  show: PropTypes.bool,
  text: PropTypes.string,
  closeCallback: PropTypes.func
};

const mapStateToProps = state => ({
  show: state.typeError.showModal,
  text: state.typeError.text
});

const mapDispatchToProps = {
  closeTypeError: actions.closeTypeError
};

export default connect(mapStateToProps, mapDispatchToProps)(ErrorModal);