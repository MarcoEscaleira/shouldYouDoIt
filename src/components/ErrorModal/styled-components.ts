import styled, { css } from "styled-components";
import { sizes, pallete } from "~/styles/_settings";

type ContainerProps = {
  isHidden: boolean;
};

export const Container = styled.div<ContainerProps>`
  display: flex;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  ${props =>
    props.isHidden &&
    css`
      visibility: hidden;
    `};
`;

export const Overlay = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  background-color: rgba(0, 0, 0, 0.5);
`;

export const Content = styled.div`
  background-color: ${pallete.fifth_light};
  color: ${pallete.white};
  width: 30rem;
  border-radius: 1rem;
  outline: none;
  padding: ${sizes.s};
  text-align: center;
  word-break: break-all;

  @media (min-width: ${sizes.desktop_breakpoint}) {
    width: 40rem;
  }
`;

export const Title = styled.h1`
  margin: 0 0 ${sizes.m} 0;
`;

export const Text = styled.p`
  font-size: ${sizes.m};
  font-weight: 300;
  margin: 0 0 ${sizes.l} 0;
  word-break: normal;
`;

export const Button = styled.button`
  background-color: ${pallete.third_dark};
  border: none;
  border-radius: 1rem;
  width: 10rem;
  font-size: 1.4rem;
  color: #ffffff;
  font-weight: 500;
  padding: ${sizes.s};
  margin: 0 auto;
`;
