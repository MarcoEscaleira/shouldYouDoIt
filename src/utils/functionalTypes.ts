export type NullaryFn<R> = () => R;
export type UnaryFn<A, R> = (a: A) => R;
export type BinaryFn<A, B, R> = (a: A, b: B) => R;
export type TernaryFn<A, B, C, R> = (a: A, b: B, c: C) => R;
export type QuadFn<A, B, C, D, R> = (a: A, b: B, c: C, d: D) => R;
export type QuinFn<A, B, C, D, F, R> = (a: A, b: B, c: C, d: D, f: F) => R;
export type SixFn<A, B, C, D, F, H, R> = (
  a: A,
  b: B,
  c: C,
  d: D,
  f: F,
  h: H
) => R;
