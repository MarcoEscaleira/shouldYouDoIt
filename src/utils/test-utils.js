import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import store from '../store';

export const renderWithRedux = (ui, initialState) => {
  return {
    ...render(
      <Provider store={store(initialState)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = ui => {
  return {
    ...render(<Router>{ui}</Router>)
  };
};

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
