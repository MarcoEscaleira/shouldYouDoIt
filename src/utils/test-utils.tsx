import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import store, { initialStoreState } from "../store";

export const renderWithRedux = (ui: JSX.Element, initialState?: {}) => {
  const state = {
    ...initialStoreState,
    ...initialState
  };
  return {
    ...render(
      <Provider store={store(state)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = (ui: JSX.Element) => ({
  ...render(<Router>{ui}</Router>)
});

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
