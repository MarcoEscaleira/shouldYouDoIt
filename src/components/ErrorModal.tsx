import React from 'react';
import { NullaryFn } from 'utils/functionalTypes';

interface Props {
  show: boolean,
  text: string,
  closeCallback: NullaryFn<void>
};

const ErrorModal: React.FC<Props> = ({ show, text, closeCallback }) => (
  <div className={`modal ${!show ? 'modal--invisible' : ''}`}>
    <div className="modal__overlay" onClick={closeCallback}>
      <div className="modal__content">
        <h3 className="modal__title">Error</h3>
        <p className="modal__body">{text}</p>
        <button className="modal__button" onClick={closeCallback}>
          Okay
        </button>
      </div>
    </div>
  </div>
);

ErrorModal.defaultProps = {
  show: false
}

export default ErrorModal;
