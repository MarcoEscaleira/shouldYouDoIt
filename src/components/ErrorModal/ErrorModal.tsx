import React from "react";
import { connect } from "react-redux";
import { actions } from "../../store/errorModal";
import { NullaryFn } from "../../utils/functionalTypes";
import {
  Container,
  Overlay,
  Content,
  Title,
  Text,
  Button
} from "./styled-components";

interface Props {
  show: boolean;
  text: string;
  closeModal: NullaryFn<void>;
}

const ErrorModal: React.FC<Props> = ({ show, text, closeModal }) => (
  <Container isHidden={!show}>
    <Overlay onClick={closeModal}>
      <Content>
        <Title>Oops!</Title>
        <Text>{text}</Text>
        <Button onClick={closeModal}>Okay</Button>
      </Content>
    </Overlay>
  </Container>
);

ErrorModal.defaultProps = {
  show: false
};

const mapDispatchToProps = {
  closeModal: actions.closeModal
};

export default connect(null, mapDispatchToProps)(ErrorModal);
