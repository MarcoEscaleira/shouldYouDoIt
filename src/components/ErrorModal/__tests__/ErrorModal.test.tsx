import React from "react";
import ErrorModal from "../ErrorModal";
import { renderWithRedux } from "~/utils/test-utils";

describe("Error Modal test suite", () => {
  it("Should render Error Modal correctly", () => {
    const { container } = renderWithRedux(
      <ErrorModal show text="Error modal text" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("Should hide Error Modal", () => {
    const { container } = renderWithRedux(<ErrorModal show={false} text="" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
