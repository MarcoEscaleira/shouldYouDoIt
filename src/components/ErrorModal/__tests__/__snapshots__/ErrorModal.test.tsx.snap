// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Modal test suite Should hide Error Modal 1`] = `
<div
  class="sc-bdVaJa jxEdDa"
>
  <div
    class="sc-bwzfXH huiBmN"
  >
    <div
      class="sc-htpNat dvDDSA"
    >
      <h1
        class="sc-bxivhb gqFfed"
      >
        Oops!
      </h1>
      <p
        class="sc-ifAKCX cyqsyD"
      />
      <button
        class="sc-EHOje gNwhWQ"
      >
        Okay
      </button>
    </div>
  </div>
</div>
`;

exports[`Error Modal test suite Should render Error Modal correctly 1`] = `
<div
  class="sc-bdVaJa ezOVdl"
>
  <div
    class="sc-bwzfXH huiBmN"
  >
    <div
      class="sc-htpNat dvDDSA"
    >
      <h1
        class="sc-bxivhb gqFfed"
      >
        Oops!
      </h1>
      <p
        class="sc-ifAKCX cyqsyD"
      >
        Error modal text
      </p>
      <button
        class="sc-EHOje gNwhWQ"
      >
        Okay
      </button>
    </div>
  </div>
</div>
`;
